<?xml version="1.0"?>
<launch>
    <arg name="trajectory" default="linear"/>
    <arg name="mav_name"   default="firefly" />
 	<arg name="x" 		   default="2.0" />
    <arg name="y"          default="2.0" />
	
	<include file="$(find rotors_gazebo)/launch/mav.launch" />

    <include file="$(find husky_gazebo)/launch/spawn_husky.launch" >
	    <arg name="x" value="$(arg x)" />
        <arg name="y" value="$(arg y)" />
        <arg name="z" value="0.3" />
		
	</include>
 
 	<include file="$(find mav_linear_mpc)/launch/mav_linear_mpc_sim.launch" />

		<node pkg="auto_landing" type="auto_landing_tracking_node" name="auto_landing_tracking_node" output="screen">
			<remap from="command_pose" 	     to="$(arg mav_name)/command/pose" />
			<remap from="husky_odometry"     to="/odometry/filtered" />
			<remap from="quad_odometry"  	 to="$(arg mav_name)/ground_truth/odometry" />
			<remap from="husky_cmd_velocity" to="/husky_velocity_controller/cmd_vel" />
			<remap from="model_state" 		 to="/gazebo/model_states" />
		</node>

        <node pkg="auto_landing" type="auto_landing_husky_node" name="auto_landing_husky_node" output="screen" args="-trajectory $(arg trajectory)">
            <rosparam command="load" file="$(find auto_landing)/cfg/husky_trajectory_params.yaml" />
            <remap from="command_velocity" to="husky_velocity_controller/cmd_vel" />
			<remap from="model_state" 		 to="/gazebo/model_states" />
        </node>
</launch>

